docs/API/API-functions.md

```diff
@@ -1558,24 +1558,29 @@ port number after it has been autoselected.
 [:arrow_up: &nbsp; Back to List of Functions & Structures](#srt-api-functions)
 
 ---  
-  
+
 ### srt_getsockopt
 ### srt_getsockflag
-```
+
+```c++
 int srt_getsockopt(SRTSOCKET u, int level /*ignored*/, SRT_SOCKOPT opt, void* optval, int* optlen);
 int srt_getsockflag(SRTSOCKET u, SRT_SOCKOPT opt, void* optval, int* optlen);
 ```
 
-Gets the value of the given socket option (from a socket or a group). 
+Gets the value of the given socket option (from a socket or a group).
 
-The first version ([`srt_getsockopt`](#srt_getsockopt)) respects the BSD socket 
-API convention, although the "level" parameter is ignored. The second version 
+The first version ([`srt_getsockopt`](#srt_getsockopt)) follows the BSD socket
+API convention, although the "level" parameter is ignored. The second version
 ([`srt_getsockflag`](#srt_getsockflag)) omits the "level" parameter completely.
 
-Options correspond to various data types, so you need to know what data type is 
-assigned to a particular option, and to pass a variable of the appropriate data 
-type. Specifications are provided in the `apps/socketoptions.hpp` file at the 
-`srt_options` object declaration.
+Options correspond to various data types (see [API-socket-options.md](#API-socket-options.md)).
+A variable `oprval` of the appropriate data type has to be passed.
+The integer value of `optlen` should originally contain the size of `optval` type provided;
+on return, it will be set to the size of the value returned.
+For most options, it will be the size of an integer. Some option, however, use `bool`, `int64_t`, `string` etc. types
```

Comment:
```suggestion
The integer value of `optlen` should originally contain the size of the `optval` type provided;
on return, it will be set to the size of the value returned.
For most options, it will be the size of an integer. Some options, however, use types `bool`, `int64_t`, `string`, etc.
```